server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,307 @@
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        switch (ctx.getRuleIndex()) {
+            case RULE_cics_perform_dump:
+                checkDump((CICSParser.Cics_perform_dumpContext) ctx);
+                break;
+            case RULE_cics_perform_endaffinity:
+                checkEndAffinity((CICSParser.Cics_perform_endaffinityContext) ctx);
+                break;
+            case RULE_cics_perform_jvmserver:
+                checkJvmServer((CICSParser.Cics_perform_jvmserverContext) ctx);
+                break;
+            case RULE_cics_perform_pipeline:
+                checkPipeline((CICSParser.Cics_perform_pipelineContext) ctx);
+                break;
+            case RULE_cics_perform_secdiscovery:
+                checkSecdiscovery((CICSParser.Cics_perform_secdiscoveryContext) ctx);
+                break;
+            case RULE_cics_perform_security:
+                checkSecurity((CICSParser.Cics_perform_securityContext) ctx);
+                break;
+            case RULE_cics_perform_shutdown:
+                checkShutdown((CICSParser.Cics_perform_shutdownContext) ctx);
+                break;
+            case RULE_cics_perform_ssl:
+                checkSsl((CICSParser.Cics_perform_sslContext) ctx);
+                break;
+            case RULE_cics_perform_statistics:
+                checkstatistics((CICSParser.Cics_perform_statisticsContext) ctx);
+                break;
+            default:
+                break;
+        }
+        checkDuplicates(ctx);
+        DUPLICATE_CHECK_OPTIONS.put(CICSLexer.DUMP, ErrorSeverity.WARNING);
+    }
+    private void checkDump(CICSParser.Cics_perform_dumpContext ctx) {
+        checkHasMandatoryOptions(ctx.DUMP(), ctx, "DUMP");
+        checkHasMandatoryOptions(ctx.DUMPCODE(), ctx, "DUMPCODE");
+        checkAllOptionsArePresentOrAbsent("TITLE, TITLELENGTH", ctx, ctx.TITLE(), ctx.TITLELENGTH());
+        checkAllOptionsArePresentOrAbsent("CALLER, CALLERLENGTH", ctx, ctx.CALLER(), ctx.CALLERLENGTH());
+    }
+    private void checkEndAffinity(CICSParser.Cics_perform_endaffinityContext ctx) {
+        checkHasMandatoryOptions(ctx.ENDAFFINITY(), ctx, "ENDAFFINITY");
+        checkHasMandatoryOptions(ctx.NETNAME(), ctx, "NETNAME");
+    }
+    private void checkJvmServer(CICSParser.Cics_perform_jvmserverContext ctx) {
+        checkHasMandatoryOptions(ctx.JVMSERVER(), ctx, "JVMSERVER");
+        checkHasExactlyOneOption("JVMTYPE or JVM or LIBERTY or OSGI", ctx, ctx.JVMTYPE(), ctx.JVM(), ctx.LIBERTY(), ctx.OSGI());
+        if (!ctx.JVM().isEmpty()) {
+            checkOptsLibertyPresent(ctx);
+            checkOptsOsgiPresent(ctx);
+            checkHasExactlyOneOption("JVMACTION or DUMP or GATHER or STACKTRACE", ctx, ctx.JVMACTION(), ctx.DUMP(), ctx.GATHER(), ctx.STACKTRACE());
+            if (!ctx.DUMP().isEmpty()) {
+                if (ctx.DUMP().size() > 1) {
+                    DUPLICATE_CHECK_OPTIONS.put(CICSLexer.DUMP, ErrorSeverity.ERROR);
```

Comment:
I think the DUMP option will have to be removed from the DUPLICATE_CHECK_OPTIONS map, and the duplicates will have to be handled manually this time in affected rules.

starting from the 2nd node do
```suggestion
checkDumpDuplicates(List<...> ..., ErrorSeverity s){

for (...)
                            throwException(
                                    s,
                                    getLocality( ...token... ),
                                    "Excessive options provided for: ",
                                    "DUMP");
}

```